contracts/Registry.sol

```diff
@@ -45,10 +50,66 @@ contract TablelandTables is
         setBaseURI(baseURI);
     }
 
-    event RunSQL(uint256 tableId, address caller, string statement);
+    event CreateTable(address caller, uint256 tableId, string statement);
+
+    function createTable(address caller, string memory statement) public {
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
+        _safeMint(caller, tokenId);
+
+        emit CreateTable(caller, tokenId, statement);
+    }
+
+    event RunSQL(
+        address caller,
+        uint256 tokenId,
+        string statement,
+        TablelandControllerLibrary.Policy policy
+    );
+
+    function runSQL(address caller, uint256 tokenId, string memory statement) public {
+        require(caller == _msgSender(), "Tables: caller must be sender"); // temp, caller must be sender (later msg.sender could be a delegate)
+        
+        TablelandControllerLibrary.Policy memory policy = _checkController(caller, tokenId);
+
+        emit RunSQL(caller, tokenId, statement, policy);
+    }
+
+    event SetController(uint256 tokenId, address controller);
 
-    function runSQL(uint256 tableId, address  controller, string memory query) public {
-        emit RunSQL(tableId, controller, query);
+    function setController(address caller, uint256 tokenId, address controller) public {
+        require(caller == _msgSender(), "Tables: caller must be sender"); // temp, caller must be sender (later msg.sender could be a delegate)
+        require(_isApprovedOrOwner(caller, tokenId), "ERC721: caller is not owner nor approved");
+
+        _controllers[tokenId] = controller;
+
+        emit SetController(tokenId, controller);
+    }
+
+    function _checkController(address caller, uint256 tokenId) private view returns (TablelandControllerLibrary.Policy memory) {
+        address controller = _controllers[tokenId];
+        if (_isContract(controller)) {
+            TablelandController c = TablelandController(controller);
+            return c.getPolicy(caller);
+        }
+        require(controller == address(0) || controller == caller, "Tables: unauthorized");
+
+        return TablelandControllerLibrary.Policy({
+            allowInsert: true,
+            allowUpdate: true,
+            allowDelete: true,
+            whereClause: "",
+            withCheck: "",
+            updatableColumns: new string[](1)
```

Comment:
Shouldn't the default be `0` instead of `1`? I think `0` creates an empty array and `1` creates an array with 1 empty string.

Reply:
ah, good catch. fixed.